raiden/connection_manager.py

```diff
@@ -287,50 +296,49 @@ def _open_channels(self):
             - This method must be called with the lock held.
         """
 
-        while True:
```

Comment:
why did we have a `while True` here?

Reply:
@andre ^ https://github.com/raiden-network/raiden/commit/f88d4c1fb55c5bd3ca36e667d26dd023901dd86c

Isn't the whole point of the connect endpoint to plan how many channels it needs to open, initiate the transactions in greenlets and then join them? Why the loop there I am not sure I understand.

Reply:
Sometimes races could happen in the middle of a loop, e.g. a channel being closed, or a new one being created by the counterpart, as it can take long waiting for a couple fo txs to be mined, which could make the view at the end of one "cycle" outdated. This loop was made to ensure that the connection manager would not only made its decisions over a static, initial "view" of the state, fire the greenlets, wait on them and return, but also it would do that as much times as required for, when the call returns, we have at least N channels opened and with funds deposited at least on our side. That was one of the reasons `test_token_networks.py` was flaky.